atom/common/extensions/atom_extensions_api_provider.cc

```diff
@@ -0,0 +1,67 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "atom/common/extensions/atom_extensions_api_provider.h"
+
+#include <string>
+
+#include "electron/buildflags/buildflags.h"
+#include "extensions/common/features/json_feature_provider_source.h"
+
+#if BUILDFLAG(ENABLE_ELECTRON_EXTENSIONS)
+#include "atom/common/extensions/api/manifest_features.h"
+#endif
+
+namespace atom {
+
+AtomExtensionsAPIProvider::AtomExtensionsAPIProvider() = default;
+AtomExtensionsAPIProvider::~AtomExtensionsAPIProvider() = default;
+
+// TODO(samuelmaddock): generate API features?
```

Comment:
We don't understand this comment. What needs to be done here?

Reply:
`AddShellAPIFeatures` is a code generated function sourced from [_api_features.json](https://cs.chromium.org/chromium/src/extensions/common/api/_api_features.json). The extensions subsystem and Chrome have their own features. We might need to do the same here.

`AddAtomManifestFeatures`, on line 32, is generated by the BUILD.gn file in this directory.